specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2022-09-08-preview/managedccf.json

```diff
@@ -0,0 +1,555 @@
+          "readOnly": true
+        }
+      }
+    },
+    "ResourceProviderOperationDefinition": {
+      "description": "Describes the Resource Provider Operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Resource provider operation name.",
+          "type": "string"
+        },
+        "isDataAction": {
+          "description": "Indicates whether the operation is data action or not.",
+          "type": "boolean"
+        },
+        "display": {
+          "description": "Details about the operations",
+          "$ref": "#/definitions/ResourceProviderOperationDisplay"
+        }
+      }
+    },
+    "ResourceProviderOperationDisplay": {
+      "description": "Describes the properties of the Operation.",
+      "type": "object",
+      "properties": {
+        "provider": {
+          "description": "Name of the resource provider.",
+          "type": "string"
+        },
+        "resource": {
+          "description": "Name of the resource type.",
+          "type": "string"
+        },
+        "operation": {
+          "description": "Name of the resource provider operation.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of the resource provider operation.",
+          "type": "string"
+        }
+      }
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "description": "Object representing ProvisioningState for Confidential Ledger.",
+      "enum": [
+        "Unknown",
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Creating",
+        "Deleting",
+        "Updating"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "ManagedCCFProperties": {
+      "type": "object",
+      "description": "Additional Managed CCF properties.",
+      "title": "ManagedCCFProperties",
+      "properties": {
+        "appName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Unique name for the Managed CCF."
+        },
+        "appUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for calling Managed CCF Service."
+        },
+        "identityServiceUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for accessing network identity."
+        },
+        "appSourceUri": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Source Uri containing mccf code"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of Ledger Resource",
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState"
+        }
+      }
+    },
+    "ManagedCCF": {
+      "type": "object",
+      "description": "Managed CCF. Contains the properties of Managed CCF Resource.",
+      "allOf": [
```

Comment:
>allOf [](http://example.com/codeflow?start=7&length=5)

All of this can be replaced by a $ref to TrackedResource definition in v3 common types.